Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -134,6 +137,14 @@ final class PurchasesOrchestrator {
         }
 
         Task {
+            #if os(iOS)
+
+            if #available(iOS 16.4, *) {
+                try? await storeMessagesHelper.deferMessagesIfNeeded()
```

Comment:
I suspect this is why it's not working in tests, we need to also check swift version here

Reply:
Tried that in https://github.com/RevenueCat/purchases-ios/pull/3252/commits/67895d1bec5102400f14a9f3722d3d4d84e83e5b but didn't work. Will try to download an older version of Xcode to try to repro this